src/mqt/ddsim/qasmsimulator.py

```diff
@@ -73,22 +74,57 @@ def target(self):
     def max_circuits(self):
         return None
 
-    def run(self, quantum_circuits: QuantumCircuit | list[QuantumCircuit], **options: dict[str, Any]) -> DDSIMJob:
+    @staticmethod
+    def _bind_parameters(
+        quantum_circuits: list[QuantumCircuit], parameter_values: Sequence[Sequence[float]] | None = None
+    ) -> list[QuantumCircuit]:
+        if parameter_values is None:
+            parameter_values = []
```

Comment:
Couldn't you just return early here?
```suggestion
        if parameter_values is None:
            return quantum_circuits
```

If you adopt the suggestion from above, you might have to write
```python
return list(quantum_circuits)
```
here.